test/core-flags/infrastructure-logging.test.js

```diff
@@ -27,4 +28,56 @@ describe('infrastructure logging related flag', () => {
         expect(stderr).toContain("Compilation 'compiler' finished");
         expect(stdout).toContain(`level: 'log'`);
     });
+
+    it('should log used default config when level is log', () => {
+        const { exitCode, stderr, stdout } = run(__dirname, ['--infrastructure-logging-level', 'log']);
+
+        expect(exitCode).toBe(0);
+        const configPath = resolve(__dirname, './webpack.config.js');
+        expect(stderr).toContain("Compilation 'compiler' starting...");
+        expect(stderr).toContain(`Using config ${configPath}`);
+        expect(stderr).toContain("Compilation 'compiler' finished");
+        expect(stdout).toContain(`level: 'log'`);
+    });
+
+    it('should log used supplied config when level is log', () => {
+        const { exitCode, stderr, stdout } = run(__dirname, [
+            '--config',
+            'webpack.cache.config.js',
+            '--infrastructure-logging-level',
+            'log',
+        ]);
+
+        expect(exitCode).toBe(0);
+        const configPath = resolve(__dirname, './webpack.cache.config.js');
+        expect(stderr).toContain(`Using config ${configPath}`);
+        expect(stdout).toContain(`level: 'log'`);
+    });
+
+    it('should log used supplied config with watch', async () => {
+        const { stderr, stdout } = await runWatch(__dirname, [
+            '--config',
+            'webpack.cache.config.js',
+            '--infrastructure-logging-level',
+            'log',
+        ]);
+
+        const configPath = resolve(__dirname, './webpack.cache.config.js');
+        expect(stderr).toContain(`Using config ${configPath}`);
+        expect(stdout).toContain(`level: 'log'`);
+    });
+
+    it('should log used supplied config with serve', async () => {
+        const { stderr, stdout } = await runWatch(__dirname, [
+            'serve',
+            '--config',
+            'webpack.cache.config.js',
+            '--infrastructure-logging-level',
+            'log',
+        ]);
+
+        const configPath = resolve(__dirname, './webpack.cache.config.js');
+        expect(stderr).toContain(`Using config ${configPath}`);
+        expect(stdout).toContain(`level: 'log'`);
+    });
```

Comment:
Wrong place for tests here, in `core-flags` we tests `flags`, put each test to `test/build/basic`/`test/watch/basic`/`test/serve/basic` directories

Reply:
moving